packages/katello/katello/hostname-change.rb

```diff
@@ -239,6 +292,107 @@ def setup_opt_parser
       @opt_parser.parse!
     end
 
+    def dns_managed?
+      @scenario_answers['foreman_proxy']['dns'] &&
+        @scenario_answers['foreman_proxy']['dns_managed'] &&
+        @scenario_answers['foreman_proxy']['dns_provider'] == 'nsupdate'
+    end
+
+    def should_update_dns?
+      dns_managed? && !@options[:skip_dns]
+    end
+
+    def answers_dns_values
+      new_vals = OpenStruct.new
+
+      new_vals.nameserver_ip = @scenario_answers['foreman_proxy']['dns_server']
+      new_vals.zone = @scenario_answers['foreman_proxy']['dns_zone']
+      new_vals.key_file = @scenario_answers['foreman_proxy']['keyfile']
+
+      new_vals.reverse_zones = [@scenario_answers['foreman_proxy']['dns_reverse']].flatten
+      run_cmd('rndc thaw')
+
+      new_vals
+    end
+
+    def nsupdate_command(dns_entries, key_file)
+      "echo -e \"#{dns_entries}\" | nsupdate -l -k #{key_file}"
+    end
+
+    def update_zone(zone, nameserver_ip, resolver)
+      commands = []
+
+      soa = resolver.getresource(zone, Resolv::DNS::Resource::IN::SOA)
+      commands << "update add #{zone} #{soa.ttl} SOA #{@new_hostname}. #{soa.rname} #{soa.serial + 1} #{soa.refresh} #{soa.retry} #{soa.expire} #{soa.minimum}"
+
+      commands << "update add #{zone}. 3600 IN NS #{@new_hostname}."
+      commands << "update delete #{zone}. IN NS #{@old_hostname}"
+
+      return commands if zone.end_with?('arpa') # don't modify A records for reverse zones
+
+      begin
+        a_record = resolver.getresource(@old_hostname, Resolv::DNS::Resource::IN::A)
+        commands << "update delete #{@old_hostname} A"
+        commands << "update add #{@new_hostname} #{a_record.ttl} A #{a_record.address}"
+      rescue Resolv::ResolvError => e
+        # This is fine
+        STDOUT.puts e
+        STDOUT.puts "no A record found; skipping"
+      end
+
+      begin
+        aaaa_record = resolver.getresource(@old_hostname, Resolv::DNS::Resource::IN::AAAA)
+        commands << "update delete #{@old_hostname} AAAA"
+        commands << "update add #{@new_hostname} #{aaaa_record.ttl} A #{aaaa_record.address}"
+      rescue Resolv::ResolvError => e
+        # This is fine
+        # TODO: Following two lines commented out until IPv6 is officially supported
```

Comment:
IPv6 is already supported in 2.1 (develop) and the installer will create the records.